modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -14,22 +14,25 @@ import org.tessellation.node.shared.ext.http4s.SnapshotOrdinalVar
 import org.tessellation.node.shared.infrastructure.snapshot.storage.SnapshotLocalFileSystemStorage
 import org.tessellation.routes.internal._
 import org.tessellation.schema.GlobalSnapshot
+import org.tessellation.schema.node.NodeState
 import org.tessellation.schema.snapshot.{Snapshot, SnapshotMetadata}
 import org.tessellation.security.Hasher
 import org.tessellation.security.signature.Signed
 
 import io.circe.Encoder
 import io.circe.shapes._
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
 import org.http4s.circe.CirceEntityEncoder
 import org.http4s.dsl.Http4sDsl
-import org.http4s.{EntityEncoder, HttpRoutes}
+import org.http4s.{EntityEncoder, HttpRoutes, Response}
 import shapeless.HNil
 import shapeless.syntax.singleton._
 
 final case class SnapshotRoutes[F[_]: Async: Hasher: KryoSerializer, S <: Snapshot: Encoder, C: Encoder](
   snapshotStorage: SnapshotStorage[F, S, C],
   fullGlobalSnapshotStorage: Option[SnapshotLocalFileSystemStorage[F, GlobalSnapshot]],
-  prefixPath: InternalUrlPrefix
+  prefixPath: InternalUrlPrefix,
+  nodeState: F[NodeState]
```

Comment:
please pass the storage instead of a single function from the store. see: `snapshotStorage` above